plugins/modules/zos_mount.py

```diff
@@ -0,0 +1,995 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2019, 2020
+# Apache License, Version 2.0 (see https://opensource.org/licenses/Apache-2.0)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community',
+}
+
+DOCUMENTATION = r"""
+---
+module: zos_mount
+author:
+    - "Rich Parker (@richp405)"
+short_description: Mount a filesystem for Unix System Services (USS)
+description:
+  - zos_mount connects an existing, mountable file to USS.
+  - Mountable file needs a valid, unique Fully Qualified Name.
+  - Target folder will be created and/or unmounted if needed
+options:
+    path:
+        description:
+            - The absolute path name onto which the file system is to be mounted
+            - The path is case sensitive, and < 1023 characters long
+        type: str
+        required: True
+    src:
+        description:
+            - The file in zos that is to be mounted
+        type: str
+        required: True
+    fstype:
+        description:
+            - The type of file system that will perform the logical mount request.
+            - The system converts the TYPE operand value to uppercase letters.
+            - The value can be up to 8 characters long.
+        type: str
+        choices:
+            - HFS
+            - zFS
+            - NFS
+            - TFS
+        required: True
+    state:
+        description:
+            - The desired status of the described mount (choice).
+            - If I(absent), this will un-mount and alse remove from fstab if provided.
+            - If I(mounted), this will mount and add to the fstab if provided.
+            - >
+                If I(mounted) and mount source is in use, left in place.
+                Will also make sure entry is in fstab, if provided.
+            - >
+                If I(unmounted) and mount source is in use, device will be unmounted.
+                fstab is not altered, even when provided.
+                Module completes successfully with I(changed=True).
+            - >
+                If I(unmounted) and mount source is not in use, no action taken.
+                Module completes successfully with I(changed=False).
+            - >
+                If I(present), if makes sure device is in fstab, if provided.
+                Returns I(changed) if fstab was rewritten.
+            - >
+                If I(remounted) the device is unmounted and mounted again.
+                Always returns I(changed=True).
+        type: str
+        choices:
+            - absent
+            - mounted
+            - unmounted
+            - present
+            - remounted
+        required: False
+        default: mounted
+    fstab:
```

Comment:
In linux, it is FileSystemTable... I added bpxprm and bpxfile as aliases, because I thought bpx_member might be a little long... I can switch it if that helps.

Reply:
We might want to get input from Kevin on this one. I think most Linux users would understand our usage of fstab in this context, but I am not sure if that applies to z/OS system programmers ... may depend on what group of users we are targeting.

Reply:
changed to persistds.  Also added .upper() on the loader to address an issue blake mentioned.

Reply:
Arguments are not capitalized...
--- They are, now.

    Do you perform...the SETOMVS SYNTAXCHECK operator command
--- Trying to get that to work

    If the src data ...
--- Agreed... I've made the mount destination and fs_type optional, and check for them in code, which should resolve this.

    fstype i...insensitive? 
--- I'll recheck... I changed it to all-caps, but I thought choice was case-sensitive.
You can reference zos_data_set for some examples of case insensitive lists of choices.

    An exception is sometimes raised when you instantiate the DataSetUtils object, most likely due to the same case issues mentioned in 1